File: exJava/go/Q12.go
package main

import "fmt"

func main() {
  array := []int{10, 8, 2, 4, 6, 7, 0};
  sort(array);
  print(array);
}

func sort(array []int) {
  for i := 0; i < len(array); i++ {
    for j:= 0; j < len(array) - 1; j++ {
      if array[j] > array[j + 1] {
        a := array[j];
        array[j] = array[j + 1];
        array[j + 1] = a;
      }
    }
  }
}

func print(array []int) {
  for i := 0; i < len(array); i++ {
    fmt.Printf("%d ", array[i]);
  }
}

File: shibaura-institute-technology/exJava/java/Q12.java
public class Q2 {

  public static void main (String[] args) {
    int[] array = {10, 8, 2, 4, 6, 7, 0};
    
    new Q2().sort(array);
    new Q2().print(array);
    //sort(array);
    //print(array);

  }

  public void sort (int[] a) {
    int l = a.length, i2, temp;
    for (int i = 0; i < l; i++) {
      for (i2 = 0; i2 < l - 1; i2++) {
  
      if (a[i2] > a[i2 + 1]) {
	      temp = a[i2];
	      a[i2] = a[i2 + 1];
	      a[i2 + 1] = temp;
        }
      }
    }
  }

  public void print (int[] a) {
    int l = a.length;
    for (int i = 0; i < l; i++) {
      System.out.println(a[i]); 
    }
  }



} 
  


